library/core/src/mem/maybe_uninit.rs

```diff
@@ -632,11 +685,12 @@ impl<T> MaybeUninit<T> {
     ///
     /// On top of that, all additional invariants of the type `T` must be
     /// satisfied, as the `Drop` implementation of `T` (or its members) may
-    /// rely on this. For example, a `1`-initialized [`Vec<T>`] is considered
-    /// initialized (under the current implementation; this does not constitute
-    /// a stable guarantee) because the only requirement the compiler knows
-    /// about it is that the data pointer must be non-null. Dropping such a
-    /// `Vec<T>` however will cause undefined behaviour.
+    /// rely on this. For example, setting a [`Vec<T>`] to an invalid but non
+    /// zero address makes it initialized (under the current implementation;
+    /// this does not constitute a stable guarantee), because the only
+    /// requirement the compiler knows about it is that the data pointer must be
+    /// non-null. Dropping such a `Vec<T>` however will cause undefined
```

Comment:
So we call the requirement "non zero" first and "non-null" later. I think it definitely should have a hyphen, I do not have a preference between "null" and "zero" here but I think "null" is more common for pointers.

Reply:
For me, both things mean the same and I vaguely remember from writing classes that you should not use the same terms repetitively, but I think technical documents are different.

Reply:
Indeed, in technical writing you absolutely *should* use the same (technical) terms when referring to the same thing. That makes it more challenging to avoid sounding repetitive, but semantic clarity is more important.